Assets/Scripts/Health/Lives.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace DN.UI
+{
+	/// <summary>
+	/// Takes care of the lives the player has to be added to Canvas
+	/// </summary>
+	public class Lives
+	{
+		public event Action<int> LifeLostEvent;
+		public event Action AllLifeLost;
+
+		public int CurrentLives { get; private set; }
+
+		private int startLives = 3;
+
+		public Lives()
+		{
+			CurrentLives = startLives;
+		}
+
+		public void LoseLife()
+		{
+			if (CurrentLives > 0)
+			{
+				CurrentLives--;
```

Comment:
Gooi dit buiten de if statement. De functie impliceert dat levens verloren worden, maar dit gebeurt alleen maar als het boven de nul is. Het is ook niet erg als er minder dan 0 levens zijn dus daar is een check voor.